openpgp/packet/packet.go

```diff
@@ -404,7 +404,9 @@ func Read(r io.Reader) (p Packet, err error) {
 		err = p.parse(contents)
 	}
 	if err != nil {
-		consumeAll(contents)
+		if _, errConsume := consumeAll(contents); errConsume != nil {
+			err = errConsume
+		}
```

Comment:
This is possible, but also a little bit up for debate - since `err != nil`, and we're now overwriting `err`, we're then essentially ignoring `err` instead of ignoring `errConsume`. It's not entirely clear which is better, and perhaps `errConsume` is more informative in some cases. But since `err` came first, I would tend to return that one. In most cases, I would guess they are the same error.

Reply:
Yes, this could be refactored. 
Contrasting to the old version, failing to error check on `consumeAll` was causing the broken key to be accepted in the first place, so they are not the same error. Given that `err != nil`, maybe the caller function is also doing things wrong.

Reply:
This `Read()` functions just documents that it returns an error and that the whole packet is consumed from input. Therefore, a caller might get it wrong, for instance, ignoring `UnknownPacketTypeError` and assuming that the `consumeAll` step was successful - I guess this was happening with the broken key.

Reply:
Added a TODO in packet/reader.go, since this escapes the scope of this branch.